runtime/domain.c

```diff
@@ -1624,6 +1624,25 @@ void caml_interrupt_all_signal_safe(void)
   }
 }
 
+static void stw_terminate_domain(caml_domain_state *domain, void *data,
+  int participating_count,
+  caml_domain_state **participating)
+{
+  if (!pthread_equal(pthread_self(), *(pthread_t *)data)) {
+    /* Ask the matching backup thread, if started, to terminate. */
+    atomic_store_release(&domain_self->backup_thread_msg, BT_TERMINATE);
```

Comment:
The code of `domain_terminate` that shuts down the backup thread also signals`domain_cond`:

https://github.com/ocaml/ocaml/blob/e1e9a7797c020a05397f52b3493688c8c7c5a4eb/runtime/domain.c#L1975-L1977

This may be necessary to do here as well, as the backup thread may be waiting on this condition (in the BT_ENTERING_OCAML case). Maybe this could be factored into an auxiliary function, say `bt_terminate`.